src/js-form-validator.ts

```diff
@@ -248,7 +256,7 @@ export class FormValidator {
       0: Not empty, but invalid.
       -1: empty
   */
-  private static validate_rule(type, value): number {
+  private static validate_rule(type, value, allowEmpty): number {
```

Comment:
@Izzmo I updated to match our coding standards, however I did not use a enum. I started on it, but couldn't see how to use an enum without it causing a breaking change. I would think our standards would be to pascal case an enum, and I would think it would be difficult to enforce pascal casing standards on `data-validate` attributes. They would need to match in order to map the `data-validate` value to the enum.